src/lib/y2storage/dialogs/guided_setup.rb

```diff
@@ -35,6 +36,41 @@ module Dialogs
     #
     # Calculates the proposal settings to be used in the next proposal attempt.
     class GuidedSetup
+      extend PartitioningFeatures
+
+      class << self
+        # Whether is is allowed to use the Guided Setup
+        #
+        # @return [Boolean]
+        def allowed?
```

Comment:
Isn't this adding one surplus indirection level?

Reply:
Yes, but as commented offline, the method `settings_editable?` is meant to only read the value from the control file and to propose a default value. I would create a new class `PartitioningSettings` with the logic to read all partitioning section from control file, proposing the expected default values for each setting. Then, current class `ProposalSettings` and this `GuidedSetup` could use it instead of reading the setting by themselves. But we are in RC3 phase.